packages/core/src/components/collapse/collapse.tsx

```diff
@@ -105,7 +106,12 @@ export enum AnimationStates {
     CLOSED,
 }
 
-export class Collapse extends AbstractPureComponent<ICollapseProps, ICollapseState> {
+export interface ICollapseSnapshot {
+    animationState?: AnimationStates;
+    height?: string;
+}
+
+class CollapseWithoutPolyfill extends AbstractPureComponent<ICollapseProps, ICollapseState, ICollapseSnapshot> {
```

Comment:
did you try the approach I suggested of polyfilling inline with `const Collapse = polyfill(class extends ...)`? I'd like to avoid having multiple symbols here, to discourage anyone from using the un-polyfilled versions (like you've done in some test suites in this PR).

Reply:
that syntax breaks several typechecks. i suggest reverting the multiple export stuff and let you apply the change you'd like to see?